### PR TITLE
Group2 filesystem mock

### PR DESCRIPTION
**Pull request by agardella**
@captain695 @cburk

### CODE REVIEW

**Comment by jfking** [comment]
What was your reasoning for using the DummyFS name instead of just naming it NativeFS or something? If it's named the latter (or something like it) we can change the implementation code without changing the code that uses it.

**Comment by agardella** [comment]
Good point, we'll change that.

File: dummy-filesystem/dummy-filesystem.cc [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+	auto iter = blockMap.find(id);
+	if(iter == blockMap.end()){
+		fputs("Error: block not found\n", stderr);
+		return -1;
+	}
+	fileName = iter->second;
+
+	//Copy to a char*, which erase and remove need
+	char *fileNameFmtd = const_cast<char*>(fileName.c_str());
+	blockMap.erase(iter);
```
Comment (efrederic) [inline dummy-filesystem/dummy-filesystem.cc]: Looks like this will remove the block from the map even if there was an error deleting the file

**Comment by kyler-m** [comment]
Not that it matters much, but stdout and stderr seem to be used interchangeably between the functions for error messages. Might want to change them all to stderr? On that note, do we want to just use the logging library for these messages instead of cerr prints?

File: dummy-filesystem/dummy-filesystem.cc [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+	// Find file size and allocate enough space
+	unsigned char* blk = (unsigned char*) malloc(sizeof(unsigned char)*blockSize);
+	if (blk == NULL) {
+		std::cout << "Error allocating memory" << std::endl;
+		return NULL;
+	}
+
+	// Copy file into buffer
+	size_t bytesRead = fread(blk, 1, blockSize, file);
+	if (bytesRead == 0) {
```
Comment (efrederic) [inline dummy-filesystem/dummy-filesystem.cc]: If we hit this case and don't free blk then it looks like there is a memory leak

File: dummy-filesystem/dummy-filesystem.cc [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+	if (!blockMap[id].empty()) {
+		return -1;
+	}
+
+	std::ostringstream oss;
+	oss << id;
+	std::string filename = "block" + oss.str() + ".txt";
+
+	blockMap[id] = filename;
+	std::ofstream myfile (filename);
```
Comment (efrederic) [inline dummy-filesystem/dummy-filesystem.cc]: "If the file cannot be opened, the stream's failbit flag is set." Should we be checking this for error opening the file?

**Comment by pelmers** [comment]
`allocateBlock` and `getBlock` take or return char pointers. Since it's c++ I would prefer we pass around either std::string or std::vector so we avoid manual memory management and out of bounds.

Also `allocateBlock` and `rmBlock` return an int where they could be more clear with bool since they are pass/fail.

And the .h file specifies methods on a class but this class is never defined in the .cc, rather we have bare functions not attached to an object.
- Move blockMap to the `private:` part of FileSystem class. 
- Define the methods by changing `int allocateBlock(long id, unsigned char* blk) {}` to `int FileSystem::allocateBlock(long id, unsigned char* blk) {}` for each of the three methods
- The test will no longer #include filesystem.cc, only filesystem.h

**Pull request merged**